presto-main-base/src/main/java/com/facebook/presto/execution/DDLDefinitionExecution.java

```diff
@@ -102,6 +108,13 @@ public DDLDefinitionExecution<?> createQueryExecution(
             checkState(preparedQuery instanceof BuiltInQueryPreparer.BuiltInPreparedQuery, "Unsupported prepared query type: %s", preparedQuery.getClass().getSimpleName());
             BuiltInQueryPreparer.BuiltInPreparedQuery builtInQueryPreparer = (BuiltInQueryPreparer.BuiltInPreparedQuery) preparedQuery;
 
+            // Run thru analyzer to get info about the update
+            QueryAnalyzer queryAnalyzer = analyzerProvider.getQueryAnalyzer();
+            AnalyzerContext analyzerContext = getAnalyzerContext(queryAnalyzer,
+                    metadata.getMetadataResolver(stateMachine.getSession()), new PlanNodeIdAllocator(), new VariableAllocator(), stateMachine.getSession(), query);
+            QueryAnalysis analysis = queryAnalyzer.analyze(analyzerContext, preparedQuery);
+            stateMachine.setUpdateInfo(analysis.getUpdateInfo());
```

Comment:
The core of the change is here - we now run DDL statements through the analyzer and use `analysis.getUpdateInfo()` to set the update info
This responsibility to set the correct/apt info about the DDL operation will now rest solely on `StatementAnalyzer`